### PR TITLE
Remove deprecated versioning decorators and helpers

### LINKED ISSUE #1424: Removed deprecated versioning helpers
These versioning helpers have been deprecated and should be removed after 2 releases. See #1381 and #1361. These should be removed in the March release. 

**Deprecations for January release**
- getRenamedFromVersion
- getRenamedFromOldName
- renamedAfter
- getAddedOn
- getRemovedOn
- addedAfter
- removedOnOrBefore

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #1424.

**TODO:**
- [x] Companion PR in `typespec-azure` (https://github.com/Azure/typespec-azure/pull/2678)
- [x] Add documentation to sprint breaking change issue (Issue #1722)

### CODE REVIEW

**Comment by tjprescott** [comment]
Can confirm that all typespec-azure tests continue to pass with these changes applied.

File: common/changes/@typespec/eslint-config-typespec/versioning-removeDeprecated_2023-03-17-18-06.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (markcowl) [inline common/changes/@typespec/eslint-config-typespec/versioning-removeDeprecated_2023-03-17-18-06.json:1]: None of the rest of these change entries should be required

File: packages/versioning/test/versioned-dependencies.test.ts [hunk 2]
```diff
@@ -402,366 +402,6 @@ describe("versioning: reference versioned library", () => {
   });
 });
 
-describe("versioning (deprecated): reference versioned library (@versionedDependency)", () => {
```
Comment (markcowl) [inline packages/versioning/test/versioned-dependencies.test.ts:405]: Do we have this same coverage in the useDependency tests?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: common/changes/@typespec/eslint-config-typespec/versioning-removeDeprecated_2023-03-17-18-06.json (hunk 1 above)
Comment (markcowl) [reply common/changes/@typespec/eslint-config-typespec/versioning-removeDeprecated_2023-03-17-18-06.json:1]: other than the versioning library

File: packages/versioning/test/versioned-dependencies.test.ts (hunk 2 above)
Comment (tjprescott) [reply packages/versioning/test/versioned-dependencies.test.ts:405]: We do.

File: common/changes/@typespec/eslint-config-typespec/versioning-removeDeprecated_2023-03-17-18-06.json (hunk 1 above)
Comment (tjprescott) [reply common/changes/@typespec/eslint-config-typespec/versioning-removeDeprecated_2023-03-17-18-06.json]: The unnecessary ones are removed.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**